src/tools/std23/views/test/repeat.test.cpp

```diff
@@ -0,0 +1,107 @@
+// include Catch2
+#include <catch2/catch_test_macros.hpp>
+
+// what we are testing
+#include "tools/std23/views.hpp"
+
+// other includes
+#include <forward_list>
+#include <list>
+#include <vector>
+#include "tools/std20/algorithm.hpp"
+#include <string_view>
+#include "tools/std20/iterator/unreachable.hpp"
+// convenience typedefs
+using namespace njoy::tools;
+
+SCENARIO( "repeat_view" ) {
+	// Use a single bound limit for all bound tests
+	int boundLimit = 4;
+
+	GIVEN( "Test repeat integer" ) {
+		int repeated = 9;
+		WHEN("Bound repeater"){
+			auto RepeatedInt = std23::views::repeat(repeated, boundLimit);
+			THEN("Check all ints in bound limit"){
+				CHECK(RepeatedInt.size() == boundLimit);
+				for (int i : RepeatedInt){
+					CHECK(i == repeated);
+				}
```

Comment:
I would like to see tests for the concepts that the view needs to satisfy:

```
        CHECK( std20::ranges::viewable_range< Range > );

        CHECK( std20::ranges::range< Range > );
        CHECK( std20::ranges::view< Range > );
        CHECK( std20::ranges::sized_range< Range > );
        CHECK( std20::ranges::forward_range< Range > );
        CHECK( std20::ranges::bidirectional_range< Range > );
        CHECK( std20::ranges::random_access_range< Range > );
        CHECK( ! std20::ranges::contiguous_range< Range > );
        CHECK( std20::ranges::common_range< Range > );
```
and tests for the remainder of the view interface. like this:
```
        CHECK( 5 == chunk.size() );

        CHECK( false == chunk.empty() );
        CHECK( true == bool( chunk ) );

        CHECK( std20::ranges::equal( equal, chunk ) );

        CHECK( equal[0] == chunk.front() );
        CHECK( equal[4] == chunk.back() );

        CHECK( 1 == chunk[0] );
        CHECK( 3 == chunk[1] );
        CHECK( 5 == chunk[2] );
        CHECK( 7 == chunk[3] );
        CHECK( 9 == chunk[4] );
```
All other views have tests like these to ensure the views function appropriately.